specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobPrivateEndpoints.json

```diff
@@ -0,0 +1,337 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2023-05-01-preview",
+    "title": "SqlManagementClient",
+    "description": "The Azure SQL Database management API provides a RESTful set of web APIs that interact with Azure SQL Database services to manage your databases. The API enables users to create, retrieve, update, and delete databases, servers, and other entities."
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/servers/{serverName}/jobAgents/{jobAgentName}/privateEndpoints": {
+      "get": {
+        "tags": [
+          "JobPrivateEndpoints"
+        ],
+        "description": "Gets a list of job agent private endpoints.",
+        "operationId": "JobPrivateEndpoints_ListByAgent",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "#/parameters/ServerNameParameter"
+          },
+          {
+            "name": "jobAgentName",
```

Comment:
The resource name parameter 'serverName' should be defined with a 'pattern' restriction.

Reply:
I could not find this pattern restriction in any of our SQL swagger files. This is not something our existing generator supports. You can see that the parent resource currently in production 'JobAgents.json' does not have any pattern restrictions. Please feel free to point me to any SQL swagger file that has this pattern restriction if I am mistaken. 

// parent resource currently live in production. 
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/sql/resource-manager/Microsoft.Sql/preview/2023-02-01-preview/JobAgents.json